sdk/core/amqp-common/src/auth/aad.ts

```diff
@@ -59,7 +59,9 @@ export class AadTokenProvider implements TokenProvider {
       );
     }
     if (credentials instanceof MSITokenCredentials) {
-      credentials.resource = Constants.aadEventHubsAudience;
+      if (!credentials.resource || credentials.resource == undefined) {
```

Comment:
The 2nd check is a bit redundant since `!credentials.resource` will also return true if `credentials.resource` is `undefined`.

Reply:
Ah yes, fixed it
Meanwhile, found this to be a little interesting - supposedly empty arrays, objects evaluate to being "truthy" but empty strings are "falsy"

Reply:
What's really crazy is that `new String()` is truthy (since it actually returns an object instead of a string primitive) but `String()` is falsy :-D